raiden_contracts/tests/test_user_deposit_contract.py

```diff
@@ -0,0 +1,119 @@
+import pytest
+from eth_tester.exceptions import TransactionFailed
+
+from raiden_contracts.constants import CONTRACTS_VERSION, UserDepositEvent
+
+
+def test_deposit(
+    user_deposit_contract,
+    custom_token,
+    get_accounts,
+):
+    (A, B) = get_accounts(2)
+    custom_token.functions.mint(30).transact({'from': A})
+    custom_token.functions.approve(user_deposit_contract.address, 30).transact({'from': A})
+
+    # deposit to A's own balance
+    user_deposit_contract.functions.deposit(A, 10).transact({'from': A})
+    assert user_deposit_contract.functions.balances(A).call() == 10
+    assert custom_token.functions.balanceOf(A).call() == 20
+
+    # increase A's deposit
+    user_deposit_contract.functions.deposit(A, 20).transact({'from': A})
+    assert user_deposit_contract.functions.balances(A).call() == 20
+    assert custom_token.functions.balanceOf(A).call() == 10
+
+    # a deposit can't be decreased by calling deposit
+    with pytest.raises(TransactionFailed):
+        user_deposit_contract.functions.deposit(A, 19).transact({'from': A})
+
+    # A deposits to the benefit of B
+    user_deposit_contract.functions.deposit(B, 10).transact({'from': A})
+    assert user_deposit_contract.functions.balances(B).call() == 10
+    assert custom_token.functions.balanceOf(A).call() == 0
+
+    # can't deposit if not enough tokens are approved
+    with pytest.raises(TransactionFailed):
+        user_deposit_contract.functions.deposit(A, 21).transact({'from': A})
```

Comment:
Not sure if this is failing for lack of balance or insufficient approval.

Reply:
I've increased the minted balance to make that unambiguous.

I didn't add a separate test for failure due to insufficient funds, since this distinction is not really relevant to the UDC and would rather be a test for the token implementation.

Reply:
The comment currently only talks about `not enough tokens are approved`.  But I'm not sure if the comment is correct.  Maybe `depositing more than the token contract allows`.

Reply:
To me the "not enough approved" and "more than allowed" have the same meaning in this context. Am I missing a subtle difference?

Reply:
"more than allowed" == ("'more than the available balance" or "more than approved").